numba_cuda/numba/cuda/codegen.py

```diff
@@ -192,7 +184,28 @@ def get_cubin(self, cc=None):
         if self.needs_cudadevrt:
             linker.add_file_guess_ext(get_cudalib('cudadevrt', static=True))
 
+    def get_cubin(self, cc=None):
+        cc = self._ensure_cc(cc)
+
+        cubin = self._cubin_cache.get(cc, None)
+        if cubin:
+            return cubin
+
+        if config.DUMP_ASSEMBLY:
+            linker = driver.Linker.new(max_registers=self._max_registers, cc=cc, additional_flags=["-ptx"])
+            self._link_all(linker, cc)
+            ptx = linker.get_linked_ptx().decode('utf-8')
+
+            if config.DUMP_ASSEMBLY:
+                print(("ASSEMBLY (AFTER LTO) %s" % self._name).center(80, '-'))
```

Comment:
If `dump_assembly=1`, the pipeline will dump PTX twice: once before LTO, and once after LTO. I think it might be helpful to keep both results. So I added `(AFTER LTO)` here for clarity.